### PR TITLE
added getXY to ball and player and getXYVelocity to movingball and mo…

### PR DESCRIPTION
**Pull request by romni**
…vingplayer, refactored engines accordingly

### CODE REVIEW

File: src/main/java/io/leonis/subra/game/data/Ball.java [hunk 1]
```diff
@@ -50,6 +50,12 @@ default double getZ() {
     return this.getState().getMean().getDouble(3, 0);
   }
 
+  /**
+   * @return The XY-position of the {@link Ball}.
+   */
+  default INDArray getXY() {
```
Comment (thumbnail) [inline src/main/java/io/leonis/subra/game/data/Ball.java]: We might want to rename this to something more conventional like `getPosition` ? Although that's as specific as `getXY()`.

File: src/main/java/io/leonis/subra/game/data/Player.java [hunk 2]
```diff
@@ -29,19 +29,26 @@ default double getX() {
...
-   * @return The Y-position coordinate of the {@link Agent}.
+   * @return The Y-position coordinate of the {@link Player}.
    */
   default double getY() {
     return this.getState().getMean().getDouble(2, 0);
   }
 
   /**
-   * @return The orientation of the {@link Agent}.
+   * @return The orientation of the {@link Player}.
```
Comment (thumbnail) [inline src/main/java/io/leonis/subra/game/data/Player.java]: Should we specify the dimensions of the orientation in this interface?

File: src/main/java/io/leonis/subra/game/data/Ball.java (hunk 1 above)
Comment (romni) [reply src/main/java/io/leonis/subra/game/data/Ball.java]: there already is a `getPosition` which returns a `(x, y, r)`

File: src/main/java/io/leonis/subra/math/spatial/GaussianPotentialField.java [hunk 3]
```diff
@@ -35,12 +36,12 @@ public INDArray getForce(final INDArray positionVector) {
         Nd4j.zeros(positionVector.rows(), 1)
             .put(0, 0,
                 (sink ? -1 : 1) *
-                    (-2d * this.origin.add(positionVector).norm2Number().doubleValue()
-                        * StrictMath.pow(this.scale, 2)
+                    (-2d * this.height * this.origin.add(positionVector).norm2Number().doubleValue()
+                        * StrictMath.pow(this.height, 2)
```
Comment (thumbnail) [inline src/main/java/io/leonis/subra/math/spatial/GaussianPotentialField.java]: Why do you use `StrictMath` instead on `Math`; even though `Math` does not guarantee bit-for-bit equal results; it is likely to be more performant. as the ~dogs~ docs say:
>  Code generators are encouraged to use platform-specific native libraries or microprocessor instructions, where available, to provide higher-performance implementations of Math methods

File: src/main/java/io/leonis/subra/game/data/Ball.java (hunk 1 above)
Comment (thumbnail) [reply src/main/java/io/leonis/subra/game/data/Ball.java]: In that case; let's change this in the future if we feel like it.

**Review by thumbnail — APPROVED** [review]
Seems good!

**Pull request merged**